src/hotspot/share/gc/shenandoah/shenandoahGeneration.cpp

```diff
@@ -236,19 +222,18 @@ void ShenandoahGeneration::reset_mark_bitmap() {
   heap->workers()->run_task(&task);
 }
 
-// The ideal is to swap the remembered set so the safepoint effort is no more than a few pointer manipulations.
-// However, limitations in the implementation of the mutator write-barrier make it difficult to simply change the
-// location of the card table.  So the interim implementation of swap_remembered_set will copy the write-table
-// onto the read-table and will then clear the write-table.
-void ShenandoahGeneration::swap_remembered_set() {
+// Swap the read and write card table pointers prior to the next remset scan.
+// This avoids the need to synchronize reads of the table by the GC workers doing
+// remset scanning, on the one hand, with the dirtying of the table by mutators
+// and by the GC workers doing remset scans, on the other.
```

Comment:
Remove "and by the the GC workers doing remset scans" from this comment, so it reads:
"This avoids the need to synchronize reads of the table by the GC workers doing
remset scanning, on the one hand, with the dirtying of the table by mutators on the other.

Reply:
@kdnilsen , when are the existing intergenerational pointers (represented by dirty cards in the read table) transferred to the write table?

Reply:
That happens during remembered set scanning.  If the remembered set scan (from the read table) finds that certain cards still hold interesting pointers, the card will be dirtied in the write table.